src/test/java/com/karankumar/bookproject/backend/entity/AuthorTest.java

```diff
@@ -35,36 +35,36 @@
 
 @IntegrationTest
 class AuthorTest {
-    private static BookService bookService;
+    private final BookService bookService;
 
     private static Book testBook1;
     private static Book testBook2;
-    private static AuthorService authorService;
+    private final AuthorService authorService;
     private static PredefinedShelf toRead;
 
-    @BeforeAll
-    public static void setup(@Autowired PredefinedShelfService predefinedShelfService,
-                             @Autowired BookService bookService,
-                             @Autowired AuthorService authorService) {
+    public AuthorTest(final @Autowired PredefinedShelfService predefinedShelfService,
+                      final @Autowired BookService bookService,
+                      final @Autowired AuthorService authorService) {
         toRead = new PredefinedShelfUtils(predefinedShelfService).findToReadShelf();
         testBook1 = createBook("How the mind works", toRead);
         testBook2 = createBook("The better angels of our nature", toRead);
 
-        AuthorTest.bookService = bookService;
-        AuthorTest.authorService = authorService;
+        this.bookService = bookService;
+        this.authorService = authorService;
 
-        resetBookService();
+        setUp();
 
-        AuthorTest.bookService.save(testBook1);
-        AuthorTest.bookService.save(testBook2);
+        this.bookService.save(testBook1);
+        this.bookService.save(testBook2);
```

Comment:
Could you please extract these lines into a method called `saveBooks()` and call this in `setUp()`? For good flow, please ensure this new `saveBooks()` method appears after `setUp()`

Reply:
All changes done and pushed! You can check them now

Reply:
I noticed you still have the `this` modifier in `setUp()`

Reply:
Since this was the only change, I've added this in for you